lib/rules/require-button-type.js

```diff
@@ -25,7 +38,7 @@ module.exports = class RequireButtonType extends Rule {
 
         let typeAttribute = attributes.find(it => it.name === 'type');
         if (!typeAttribute) {
-          return this.logNode({ node, message: ERROR_MESSAGE });
+          return this.logNode({ node, message: ERROR_MESSAGE, fixer: addDefaultAttribute });
         }
```

Comment:
I'm wondering if it would be easier to have a `this.shouldFix` property on the rule and apply the fix directly here, instead of having to use another visitor for it?

Reply:
Ya, agree. As it stands this fixer seems like it would adding a `type="button"` attribute for **EVERY** element that exists in the source.  It is pretty common to have complex logic to decide if a given node is actually an error to be logged, and that same logic will be needed for the fixer...